File: 80. Remove Duplicates from Sorted Array II/Go/v1/removeDuplicates.go
package v1

func removeDuplicates(nums []int) int {
	if len(nums) <= 2 {
		return len(nums)
	}

	unique := 2

	for i := 2; i < len(nums); i++ {
		if nums[i] != nums[unique-2] {
			nums[unique] = nums[i]
			unique++
		}
	}
	return unique
}

File: 80. Remove Duplicates from Sorted Array II/JS/v1/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let twice = false;
  let unique2 = 1;
  let left = 1;
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] === nums[left]) {
      if (!twice) {
        nums[left] = nums[i];
        left++;
        twice = true;
        unique2++;
      }
    } else {
      nums[left] = nums[i];
      left++;
      twice = false;
      unique2++;
    }
  }

  return unique2;
};
